src/dswx_sar/save_mgrs_tiles.py

```diff
@@ -155,37 +205,39 @@ def find_intersecting_burst_with_bbox(ref_bbox,
 
     overlapped_rtc_dir_list = []
     for input_dir in input_rtc_dirs:
-        copol_rtc_path_iter = glob.iglob(f'{input_dir}/*_VV*.tif')
-        copol_rtc_path = next(copol_rtc_path_iter)
-
-        if not copol_rtc_path:
-            copol_rtc_path_iter = glob.iglob(f'{input_dir}/*_HH*.tif')
-            copol_rtc_path = next(copol_rtc_path_iter)
-
-        with rasterio.open(copol_rtc_path) as src:
-            epsg_code = int(src.crs.data['init'].split(':')[1])
-
-            # Get bounds of the raster data
-            left, bottom, right, top = src.bounds
-
-            # Reproject to EPSG 4326 if the current EPSG is not 4326
-            if epsg_code != ref_epsg:
-                left, bottom, right, top = \
-                    transform_bounds(src.crs,
-                                     {'init': f'EPSG:{ref_epsg}'},
-                                     left,
-                                     bottom,
-                                     right,
-                                     top)
-            rtc_polygon = Polygon([(left, bottom),
-                                   (left, top),
-                                   (right, top),
-                                   (right, bottom)])
-
-        # Check if bursts intersect the reference polygon
-        if ref_polygon.intersects(rtc_polygon) or \
-           ref_polygon.overlaps(rtc_polygon):
-            overlapped_rtc_dir_list.append(input_dir)
+        copol_file_list = []
+        for copol in dswx_s1_pol_dict['CO_POL']:
+            copol_rtc_path = glob.glob(f'{input_dir}/*_{copol}*.tif')
+            copol_file_list.extend(copol_rtc_path)
```

Comment:
List comprehension version with more than1 glob call and filtering done in glob.
```suggestion
        copol_file_list = [f for pol in dswx_s1_pol_dict['CO_POL']
                           for f in glob.glob(f'{input_dir}/*_{copol}*.tif')]
```

List comprehension version with single glob call and filtering done to all tif files.
```suggestion
        copol_file_list = [f for f in glob.glob(f'{input_dir}/*.tif')
                           if any(pol in f
                                  for pol in dswx_s1_pol_dict['CO_POL'])]
```